csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -78,6 +78,7 @@ public void Remove(Connection connection)
             bool hasMore,
             string connectionId,
             IReadOnlyList<IConnector> excludedConnectors,
+            bool preferNonSecure,
```

Comment:
Re: secure vs preferNonSecure

A connection is either secure=true or secure=false.

However, when establishing a connection, preferNonSecure (currently bool, soon enum) makes more sense and should be used consistently. When `preferNonSecure` is `NonSecure.Never` (false for now), you always get back a secure connection. Otherwise, you may get back a secure=true or a secure=false connection.

Reply:
Would be nice to add a doc-comment that describes the parameters of this method, such as the non-obvious hasMore. And I suspect preferNonSecure should not be the trailing parameter.

Reply:
`hasMore` gone in my retry loop update PR and most of the outgoing connection factory is updated there